icon_validator/rules/plugin_validators/runtime_validator.py

```diff
@@ -0,0 +1,56 @@
+import os
+import glob
+
+from icon_validator.rules.validator import KomandPluginValidator
+from icon_validator.exceptions import ValidationException
+
+
+class RuntimeValidator(KomandPluginValidator):
+
+    @staticmethod
+    def validate_setup(spec):
+        if "setup.py" in os.listdir(spec.directory):
+            with open(f"{spec.directory}/setup.py", "r") as file:
+                setup_str = file.read().replace("\n", "")
+
+                if "komand" in setup_str and "insightconnect-plugin-runtime" not in setup_str:
+                    raise ValidationException("Komand is no longer used in setup.py. "
+                                              "Use insightconnect-plugin-runtime instead.")
+
+    @staticmethod
+    def validate_imports(spec):
+        for root, dirs, files in os.walk(spec.directory):
+            for file in files:
+                if file.endswith(".py"):
+                    with open(os.path.join(root, file), "r") as open_file:
+                        file_str = open_file.read().replace("\n", "")
+
+                        if "import komand" in file_str or "FROM komand" in file_str:
+                            raise ValidationException(f"Komand import found in {str(os.path.join(root, file))}. "
+                                                      "Komand is no longer used here. "
+                                                      "Use insightconnect-plugin-runtime instead.")
+
+    @staticmethod
+    def validate_caching(spec):
+        if spec.spec_dictionary().get("cloud_ready") is True:
+            actions_path = glob.glob(f"{spec.directory}/*/actions")
+            for root, dirs, files in os.walk(actions_path[0]):
+                for file in files:
+                    with open(os.path.join(root, file), 'r') as open_file:
+                        file_str = open_file.read().replace("\n", "")
+
+                        if "cache" in file_str:
+                            raise ValidationException(f"Cloud ready plugins cannot contain caching. "
+                                                      f"Update {str(os.path.join(root, file))}.")
+
+    def validate(self, spec):
+        latest_images = ["rapid7/insightconnect-python-3-38-plugin",
```

Comment:
I'm wondering if we need to check the reverse as well.  If `insightconnect-plugin-runtime` in setup.py then make sure the proper base image is being used.  If not, it will be downloaded and used with the old komand based docker images.

Reply:
Hmm, well in theory this is already checked. We perform this validation only when the latest image is being used, and if so, then we check for `insightconnect-plugin-runtime` in setup.py. So it could be enough as-is?

Reply:
I'm thinking the other way around. insightconnect-plugin-runtime is in setup.py but the Dockerfile isn't update.  Don't think we hit that scenario do we?

Reply:
I get what you mean, I'm just trying to envision how much of an issue that is given the current validation, and whether we'd want a separate function for checking that.

Reply:
I think it is very possible someone updates setup.py and forgets to update the Dockerfile just like the other way around which you are already checking for.  If we don't have a validator it wouldn't fail since even the 3.7 base image would download insightconnect-plugin-runtime from pypi and then we would have no clue that is a problem until a potential plugin runtime error.